R/forecast.R

```diff
@@ -99,48 +107,105 @@ as_forecast_generic <- function(data,
 }
 
 
-#' @rdname as_forecast
+#' @title Create a `forecast` object for binary forecasts
+#' @description
+#' Create a `forecast` object for binary forecasts. See more information on
+#' forecast types and expected input formats by calling `?`[as_forecast()].
 #' @export
+#' @inheritParams as_forecast
+#' @seealso [as_forecast()], [as_forecast_point()], [as_forecast_binary()],
+#'   [as_forecast_sample()], [as_forecast_quantile()]
 #' @importFrom cli cli_warn
-as_forecast_point <- function(data,
-                              forecast_unit = NULL,
-                              observed = NULL,
-                              predicted = NULL,
-                              model = NULL) {
+#' @keywords check-forecasts
+as_forecast_binary <- function(data,
+                               forecast_unit = NULL,
+                               observed = NULL,
+                               predicted = NULL,
+                               model = NULL) {
   data <- as_forecast_generic(data, forecast_unit, observed, predicted, model)
-  data <- new_forecast(data, "forecast_point")
+  data <- new_forecast(data, "forecast_binary")
   assert_forecast(data)
   return(data)
 }
 
 
-#' @rdname as_forecast
+#' @title Create a `forecast` object for point forecasts
+#' @description
+#' Create a `forecast` object for point forecasts. See more information on
+#' forecast types and expected input formats by calling `?`[as_forecast()].
+#' @inherit as_forecast params
+#' @param ... Unused
+#' @seealso [as_forecast()], [as_forecast_point()], [as_forecast_binary()],
+#'   [as_forecast_sample()], [as_forecast_quantile()]
+#' @export
+#' @keywords check-forecasts
+as_forecast_point <- function(data, ...) {
+  UseMethod("as_forecast_point")
+}
+
+
+#' @rdname as_forecast_point
 #' @export
 #' @importFrom cli cli_warn
-as_forecast_binary <- function(data,
-                               forecast_unit = NULL,
-                               observed = NULL,
-                               predicted = NULL,
-                               model = NULL) {
+as_forecast_point.default <- function(data,
```

Comment:
do we think longer term this should be the default or this should be a method for a data frame? I don't think it matters that much or should hold up this PR (then all the methods could have s3 defaults + a data frame method right now without waiting for converters etc).

Reply:
I briefly had the same thought. No strong opinion 🤷 
We'd still have to convert from `data.frame` to `data.table` though

Reply:
yeah good point. I don't feel strongly the benefit is anything that isn't just a dataframe can have an error from the default method vs relying on the input checks written into this approach.